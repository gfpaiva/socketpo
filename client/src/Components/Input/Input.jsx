import React from 'react';
import PropTypes from 'prop-types';

import './Input.scss';

const Input = ({ className, ...props }) => (
	<input
		className={
		`${className ? `${className} ` : ''}
		input`}
		{...props}
	/>
);

Input.propTypes = {
	className: PropTypes.string,
	type: PropTypes.string.isRequired
};

export default Input;

