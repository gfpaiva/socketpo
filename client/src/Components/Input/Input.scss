.input {
	display: block;
	width: 80%;
	padding: .5em;
	margin: .5em auto;
	border: none;
	background-color: rgba(255, 255, 255, .5);
}
