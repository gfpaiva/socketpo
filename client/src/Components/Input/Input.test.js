import React from 'react';
import { mount } from 'enzyme';
import Input from './Input';

describe('<Input />', () => {
	it('should mount properly with class and props', () => {
		const wrapper = mount(
			<Input
				className="test"
				type="text"
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});
});
