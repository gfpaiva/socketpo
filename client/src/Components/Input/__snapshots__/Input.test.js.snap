// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should mount properly with class and props 1`] = `
<Input
  className="test"
  type="text"
>
  <input
    className="test 
		input"
    type="text"
  />
</Input>
`;
